pandas/io/xml.py

```diff
@@ -1119,6 +1125,15 @@ def read_xml(
     """
     check_dtype_backend(dtype_backend)
 
+    if isinstance(path_or_buffer, str) and "\n" in path_or_buffer:
```

Comment:
Is `\n` a reliable way detect if it's literal xml?

Reply:
I think you can use `is_file_like` (and in your other PRs)

Reply:
@mroeschke Ah thanks for letting me know about `is_file_like`!

Are we okay with updating the detection logic to include both `is_file_like` and the check for "\n"? `is_file_like doesn't help differentiate between raw xml input and a url so that's where checking for "\n" could be helpful.

Reply:
You can use `is_url` to detect urls.